kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/store/DbWireRecordStore.java

```diff
@@ -224,8 +224,8 @@ private void truncate(final int noOfRecordsToKeep) {
                     this.dbHelper.execute(MessageFormat.format(SQL_TRUNCATE_TABLE, sqlTableName));
                 } else {
                     logger.info(message.partiallyEmptyingTable(sqlTableName));
-                    this.dbHelper
-                            .execute(MessageFormat.format(SQL_DELETE_RANGE_TABLE, sqlTableName, noOfRecordsToKeep));
+                    this.dbHelper.execute(MessageFormat.format(SQL_DELETE_RANGE_TABLE, sqlTableName,
+                            Integer.toString(noOfRecordsToKeep)));
```

Comment:
You could also do the following:

```java
private static final String SQL_DELETE_RANGE_TABLE = "DELETE FROM {0} WHERE rownum() <= (SELECT count(*) FROM {0}) - {1,number,#};";
```

This way you don't have to stringify it again and this is also suggested by message formatter.